src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSslVersion.cs

```diff
@@ -0,0 +1,36 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class OpenSsl
+    {
+        private static Version s_opensslVersion;
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_SSLEayVersion")]
+        private static extern string OpenSslVersionDescription();
+
+        internal static Version OpenSslVersion
+        {
+            get
+            {
+                if (s_opensslVersion == null)
+                {
+                    const string OpenSSL = "OpenSSL ";
+
+                    // Skip OpenSSL part, and get the version string of format x.y.z
+                    if (!Version.TryParse(OpenSslVersionDescription().Substring(OpenSSL.Length, 5), out s_opensslVersion))
```

Comment:
Nit: it doesn't matter for now as this is only used in tests, but if this is ever used in src, this is a good place to use spans, e.g. using `.AsReadOnlySpan().Slice` instead of `.Substring`.